runtime/compilers/rillv1/parser.go

```diff
@@ -389,8 +398,17 @@ func (p *Parser) reparseExceptRillYAML(ctx context.Context, paths []string) (*Di
 
 		// Check if path is .env and clear it (so we can re-parse it)
 		if isDotEnv {
-			modifiedDotEnv = true
-			p.DotEnv = nil
+			oldDotEnv := p.DotEnv
+			err := p.parseDotEnv(ctx, checkPaths[i])
+			if err == nil {
+				modifiedDotEnv = !maps.Equal(p.DotEnv, oldDotEnv)
+			} else {
+				// any error means .env is under change
+				modifiedDotEnv = true
+			}
+			if modifiedDotEnv {
+				p.DotEnv = nil
+			}
 		}
```

Comment:
1. I think it shouldn't be added to `parsePaths` if it is not modified? Could move it up to around line 377 and call `continue` if it's not modified.
2. Add a comment about why we're checking this (otherwise in some future refactor, someone may not know the context and revert it)